### DISCUSSION TITLE
How to Deploy my app using serverActions

### DISCUSSION THREAD

**Discussion by bolisteward** [vercel/next.js#55698] [Help]
### Summary

I develop a small form using server actions to send the data to MySQL using a PHP endpoint to insert the data.
I want to upload my app to the cpanel but I get an error when I run the build with the output export in the next.config.js file.

What do I have to do to deploy my application to cpanel?
Do I have to separate the server action to deploy using node.js server and static file?

Please help me because I tried many things.

action.ts file 

export async function POSTData(dataAplicante: FormData) {
  try {
    const urlIMG = await saveFile(
      dataAplicante.get("foto") as File,
      dataAplicante.get("apellidos")?.toString()!,
      dataAplicante.get("id_aplicante")? dataAplicante.get("id_aplicante")?.toString()! : ""
    );

    const jsondata = createJSONdata(dataAplicante.get("action") as string, dataAplicante, urlIMG);

    const response = await fetch("https://domain.com.ec/register/", {
      method: "POST",
      headers: {
        "Content-Type": "application/json", // Establecer el tipo de contenido a JSON
      },
      body: jsondata,
    });

    if (!response.ok) {
      console.error("Error en la solicitud:", response.status);
      throw new Error("Error en la solicitud");
    }

    const data = await response.json();

    return {
      mensaje_ap: data.mensaje,
      id_aplicante: data.id,
    };
  } catch (e) {
    throw new Error("Internal Server Error");
  }
}

### Additional information

```js
I found the standalone output option, but how do I implement it with the static files? or how do I connect them?
```

**Comment by icyJoseph** [comment]
The static HTML export output mode, does not create a server, which is necessary for server actions because they are:

> Create a Server Action by defining an asynchronous function with the ["use server"](https://react.dev/reference/react/use-server) directive at the top of the function body. "use server" ensures this function is only ever executed on the server.

So we have a catch-22, you build using export mode, which doesn't create a next-server, but then you use server actions, which do require a next-server.
